app/src/main/java/com/github/onlynotesswent/ui/overview/editnote/Comments.kt

```diff
@@ -8,100 +8,149 @@ import androidx.compose.foundation.layout.Row
 import com.github.onlynotesswent.model.user.UserViewModel
+import com.github.onlynotesswent.ui.common.ThumbnailDynamicPic
 import com.github.onlynotesswent.ui.navigation.NavigationActions
 import com.github.onlynotesswent.ui.navigation.Screen
-import java.text.SimpleDateFormat
-import java.util.Locale
+import com.github.onlynotesswent.ui.theme.Typography
 
+/**
+ * Screen for viewing and editing comments on a note.
+ * @param navigationActions Actions used for navigating between screens.
+ *  @param noteViewModel The ViewModel for notes.
+ *  @param userViewModel The ViewModel for users.
+ *  @param fileViewModel The ViewModel for files.
+ *
+ */
 @Composable
 fun CommentsScreen(
     navigationActions: NavigationActions,
     noteViewModel: NoteViewModel,
-    userViewModel: UserViewModel
+    userViewModel: UserViewModel,
+    fileViewModel: FileViewModel
 ) {
-  val note by noteViewModel.selectedNote.collectAsState()
-  val currentUser by userViewModel.currentUser.collectAsState()
-  var updatedComments by remember { mutableStateOf(note!!.comments) }
-  Scaffold(
-      floatingActionButton = {
-        AddCommentButton(
-            currentUser = currentUser,
-            note = note,
-            updatedComments = updatedComments,
-            onCommentsChange = { updatedComments = it })
-      },
-      modifier = Modifier.testTag("commentsScreen"),
-      topBar = {
-        EditNoteTopBar(
-            title = stringResource(R.string.comments),
-            titleTestTag = "commentsTitle",
-            noteViewModel = noteViewModel,
-            navigationActions = navigationActions,
-            onClick = {
-              val commentsNotEmpty = updatedComments.commentsList.filter { it.content.isNotEmpty() }
-              noteViewModel.updateNote(
-                  note = note!!.copy(comments = Note.CommentCollection(commentsNotEmpty)))
-            })
-      },
-      bottomBar = {
-        EditNoteNavigationMenu(
-            navigationActions = navigationActions,
-            selectedItem = Screen.EDIT_NOTE_COMMENT,
-            onClick = {
-              val commentsNotEmpty = updatedComments.commentsList.filter { it.content.isNotEmpty() }
-              noteViewModel.updateNote(
-                  note!!.copy(comments = Note.CommentCollection(commentsNotEmpty)))
-              noteViewModel.getNoteById(note!!.id)
-            })
-      }) { paddingValues ->
+    val note by noteViewModel.selectedNote.collectAsState()
+    var noteUpdated by remember { mutableStateOf(note) }
+    val currentUser by userViewModel.currentUser.collectAsState()
+    val updatedComments by remember { derivedStateOf {  (noteUpdated!!.comments) }}
+
+    LaunchedEffect(Unit) {
+        while (true) {
+            kotlinx.coroutines.delay(1000L)
+            // Delay for 1 second to not saturate firestore but also to not get too much lag
+            if (note != null) {
+                noteViewModel.getNoteById(note!!.id, { noteUpdated = it })
```

Comment:
There could be a small synchronisation issue with the local and database note. Imagine you add a new comment and you press the button to save this new note with the new comment. Now imagine the request takes time so your new note with the new comment hasn't been still saved in the database and the 1 second is up so the `LaunchedEffect` calls `getNoteById` so it will retrieve a note which doesn't have the new comment and will assign it to `noteUpdated`. This could also cause another issue which I won't go into detail. One possible way to solve this is by merging the local and the database note before each time you update the local or database note, however I am not sure if it is the best solution. Due to our time constraint It is understandable if you do not solve this issue so I will let you decide.

Reply:
It should be fine as if it was saved to the database, the old note will be sent be then you will send you new note and then receive the old note. I had this issue once and it just causes a flicker. Why the problem does not happen is because the fact of sending was done with the correct note. What I think will happen is: old Note ->  I push new Comment -> I receive an old note (my comment disappeared) -> I receive the correct comment. This is because a stale note is not saved unless explicitly done. However to have this problem if you save a comment in the 1 second delay of fetching notes (that I set to not go over our free plan) then you can have a problem with overriding an old comment otherwise it should be fine). However for this issue that can arise when two people comment at the same time in less than a second interval I do not have time to fix this issue.